### DISCUSSION TITLE
Ability to manipulate webhook response? (Two-way webhooks)

### DISCUSSION THREAD

**Discussion by arladmin** [Budibase/budibase#1202] [Ideas]
It seems that this ability is currently missing.

Currently, whenever a webhook is called, only a standard sucess/failure response is received.

A very important and frequent use-case is to receive relevant data in the webhook response.
e.g. Automation flow: 
1. Trigger- Webhook
2. Logic- Check incoming data, apply some logic
3. Action- Fetch some data from Tables
4. Action- Apply business logic (in the form of (multiple) javascript functions) to the fetched data
5. Action- Return the resulting data as the response to the triggering webhook

^ Effectively, this becomes a custom API feature, which is great for several business scenarios in all domains.

I understand this might require some overhaul as the current webhook/trigger system seems to be designed only for one-way communication/perspective.

**Comment by arladmin** [comment]
Hey @joebudi, any thoughts about this?

**Reply by mike12345567** [reply]
Hi @arladmin this is a good idea - right now our current implementation of the webhook trigger wouldn't entirely support this, but I can think of a few ways we might go about this. I can see this being rolled into future work on automations as if we are already making changes to automations it wouldn't be that much extra work and I can definitely imagine some great use cases for it.

**Reply by basteks** [reply]
Hi there ! Sorry for digging this thread, but I just discovered Budibase which sounds pretty promising according to my use cases, and find this topuc because it's a feature I definitely need. I looked at the webhooks documentation nad didn't find anything about that. I assume it means that it has not been developed yet. Is this planned ?

Bests,
Benjamin
